build/Common.prod.props

```diff
@@ -72,10 +72,6 @@
       <DefineConstants>$(DefineConstants);EXPOSE_EXPERIMENTAL_FEATURES</DefineConstants>
     </PropertyGroup>
 
-    <ItemGroup Condition="'$(ExposeExperimentalFeatures)' != 'true'">
```

Comment:
@CodeBlanch - Why was this needed earlier?

Reply:
If you look at one of the project files on the "before" side of the diff they have things like this:

```xml
    <!-- Note: When '$(ExposeExperimentalFeatures)' == 'false' these links are
    NOT required because this project sees API + SDK internals -->
    <Compile Include="$(RepoRoot)\src\Shared\Guard.cs" Link="Includes\Guard.cs" RequiresExposedExperimentalFeatures="true" />
```

Before you could conditionally link things into projects based on whether or not experimental apis are being exposed `public` or `internal`. This PR is making `InternalsVisibleTo` a consistent thing across builds so the conditional linking is no longer needed.